docs/rototiller_task_reference.md

```diff
@@ -159,3 +162,112 @@ produces:
     command_name --switch --option opt arg arguments go last
 
     # what do you think ARG_OVERRIDE2 does?
+
+##Relationship between add_env and parameter name
+Any parameter or task can call the `add_env` method.  The context of the `add_env` method call and additional method calls determine its behavior.  The only location where `add_env` is not available is under an existing `add_env` method call.  
+  
+###A user (Task Author) wants the rake task consumer to provide input
+
+Stopping and asking for an Environment variable that represents an argument to an --option
+&nbsp;
+
+    desc "a command that will stop"
+    rototiller_task :stop do |task|
+      task.add_command do |cmd|
+        cmd.name = 'echo command_name'
+        cmd.add_option do |o|
+          o.name = '--option'
+          o.add_argument do |arg|
+            arg.add_env({ :name => 'ARG_OVERRIDE', :message => 'message at the env for argument' })
+          end
+        end
+      end
+    end
+
+
+Stopping and asking for an Environment variable that is not related to a command
+&nbsp;
+
+    desc "a command that will stop"
+    rototiller_task :stop do |task|
+      task.add_command do |cmd|
+      
+        cmd.add_env({ :name => 'ARG_OVERRIDE' })
```

Comment:
i think you want this at the task level?  won't this get a default of "echo command_name"?